core/src/action.rs

```diff
@@ -158,10 +161,10 @@ pub enum Action {
 
     /// get links from entry address and link_type name
     /// Last string is the stringified process unique id of this `hdk::get_links` call.
-    GetLinks(GetLinksKey),
+    GetLinks((GetLinksKey, Option<CrudStatus>, GetLinksNetworkQuery)),
     GetLinksTimeout(GetLinksKey),
-    RespondGetLinks((QueryEntryData, Vec<(Address, CrudStatus)>, String, String)),
-    HandleGetLinksResult((Vec<(Address, CrudStatus)>, GetLinksKey)),
+    RespondGetLinks((QueryEntryData, GetLinksNetworkResult, String, String)),
+    HandleGetLinksResult((GetLinksNetworkResult, GetLinksKey)),
```

Comment:
I would love to see these actions and the corresponding action creators and reducers in the network module to not know about links at all and just operate on the generic level of queries.